pkg/deployments/tasks/20221101-l2-gauge-checkpointer/test/task.fork.ts

```diff
@@ -0,0 +1,234 @@
+import hre, { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { Contract } from 'ethers';
+
+import { BigNumber, fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { GaugeType } from '@balancer-labs/balancer-js/src/types';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+
+import { describeForkTest } from '../../../src/forkTests';
+import Task, { TaskMode } from '../../../src/task';
+import { getForkedNetwork } from '../../../src/test';
+import { impersonate } from '../../../src/signers';
+
+// This block number is before the manual weekly checkpoint. This ensures gauges will actually be checkpointed.
+describeForkTest('L2GaugeCheckpointer', 'mainnet', 15272610, function () {
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+  let L2GaugeCheckpointer: Contract;
+  let vault: Contract, authorizer: Contract, authorizerAdaptor: Contract;
+
+  let task: Task;
+
+  const GOV_MULTISIG = '0x10A19e7eE7d7F8a52822f6817de8ea18204F2e4f';
+
+  // Gauges that are NOT killed.
+  const polygonRootGauges: [address: string, expectedCheckpoints: number][] = [
+    ['0xa5a0b6598b90d214eaf4d7a6b72d5a89c3b9a72c', 1],
+    ['0x88d07558470484c03d3bb44c3ecc36cafcf43253', 1],
+    ['0xfbf87d2c22d1d298298ab5b0ec957583a2731d15', 1],
+    ['0xc6fb8c72d3bd24fc4891c51c2cb3a13f49c11335', 1],
+    ['0x5a3970e3145bbba4838d1a3a31c79bcd35a16a9e', 1],
+    ['0xc3bb46b8196c3f188c6a373a6c4fde792ca78653', 1],
+    ['0xa80d514734e57691f45af76bb44d1202858fd1f0', 1],
+    ['0x211c27a32e686659566c3cee6035c2343d823aab', 1],
+    ['0x397649ff00de6d90578144103768aaa929ef683d', 1],
+    ['0xd27cb689083e97847dc91c64efc91c4445d46d47', 1],
+    ['0xf01541837cf3a64bc957f53678b0ab113e92911b', 1],
+    ['0xead3c3b6c829d54ad0a4c18762c567f728ef0535', 1],
+    ['0xcf5938ca6d9f19c73010c7493e19c02acfa8d24d', 1],
+    ['0xd13a839bb48d69a296a1fa6d615b6c39b170096b', 2],
+  ];
+
+  const arbitrumRootGauges: [address: string, expectedCheckpoints: number][] = [
+    ['0xF0ea3559Cf098455921d74173dA83fF2f6979495', 1],
+    ['0xB0de49429fBb80c635432bbAD0B3965b28560177', 1],
+    ['0x359EA8618c405023Fc4B98dAb1B01F373792a126', 1],
+    ['0xc77E5645Dbe48d54afC06655e39D3Fe17eB76C1c', 1],
+    ['0x899F737750db562b88c1E412eE1902980D3a4844', 1],
+    ['0x6cb1A77AB2e54d4560fda893E9c738ad770da0B0', 1],
+    ['0x6823DcA6D70061F2AE2AAA21661795A2294812bF', 1],
+    ['0xACFDA9Fd773C23c01f5d0CAE304CBEbE6b449677', 1],
+    ['0x68EBB057645258Cc62488fD198A0f0fa3FD6e8fb', 1],
+  ];
```

Comment:
Some of these might need updating

Reply:
Done; I've added the txs for last week's update too.

The list in the test matches the gauges checkpointed manually except for one, which is added for completion.

I double checked the rogue gauge, and it was probably skipped intentionally. Its relative weight for the 2 periods that it gets checkpointed in the test was 0.